irr/src/CGLTFMeshFileLoader.cpp

```diff
@@ -385,48 +410,114 @@ static std::vector<u16> generateIndices(const std::size_t nVerts)
  * Load up the rawest form of the model. The vertex positions and indices.
  * Documentation: https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#meshes
  * If material is undefined, then a default material MUST be used.
-*/
-void SelfType::MeshExtractor::loadMesh(
+ */
+void SelfType::MeshExtractor::deferAddMesh(
 		const std::size_t meshIdx,
-		ISkinnedMesh::SJoint *parent) const
-{
-	for (std::size_t pi = 0; pi < getPrimitiveCount(meshIdx); ++pi) {
-		const auto &primitive = m_gltf_model.meshes->at(meshIdx).primitives.at(pi);
-		auto vertices = getVertices(primitive);
-		if (!vertices.has_value())
-			continue; // "When positions are not specified, client implementations SHOULD skip primitive’s rendering"
-
-		// Excludes the max value for consistency.
-		if (vertices->size() >= std::numeric_limits<u16>::max())
-			throw std::runtime_error("too many vertices");
-
-		// Apply the global transform along the parent chain.
-		transformVertices(*vertices, parent->GlobalMatrix);
-
-		auto maybeIndices = getIndices(primitive);
-		std::vector<u16> indices;
-		if (maybeIndices.has_value()) {
-			indices = std::move(*maybeIndices);
-			checkIndices(indices, vertices->size());
-		} else {
-			// Non-indexed geometry
-			indices = generateIndices(vertices->size());
-		}
+		const std::optional<std::size_t> skinIdx,
+		CSkinnedMesh::SJoint *parent)
+{
+	m_mesh_loaders.emplace_back([=] {
+		for (std::size_t pi = 0; pi < getPrimitiveCount(meshIdx); ++pi) {
+			const auto &primitive = m_gltf_model.meshes->at(meshIdx).primitives.at(pi);
```

Comment:
not important: I bet a few indentation levels can be saved here.

Reply:
you're right, there was a function to be extracted lurking here